trieste/utils/multi_objectives.py

```diff
@@ -0,0 +1,225 @@
+# Copyright 2020 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains synthetic multi-objective functions, useful for experimentation.
+"""
+from __future__ import annotations
+
+import math
+from abc import ABC, abstractmethod
+from collections.abc import Callable
+from functools import partial
+
+import tensorflow as tf
+
+from ..type import TensorType
+
+
+class MultiObjectiveTestProblem(ABC):
+    """
+    Base class for synthetic multi-objective test functions.
+    Prepares the synthetic function and generates pareto optimal points.
+    The latter can be used as a reference of performance measure of certain
+    multi-objective optimization algorithms.
+    """
+
+    @property
+    @abstractmethod
+    def dim(self):
+        """
+        The input dimensionality of the test function
+        """
+
+    @property
+    @abstractmethod
+    def bounds(self):
+        """
+        The input space bounds of the test function
+        """
+
+    @abstractmethod
+    def objective(self) -> Callable[[TensorType], TensorType]:
+        """
+        Get the synthetic test function.
+        :return: A callable synthetic function
+        """
+
+    @abstractmethod
+    def gen_pareto_optimal_points(self, n: int, seed=None) -> tf.Tensor:
+        """
+        Generate `n` Pareto optimal points.
```

Comment:
There should be a blank line before the :param here (and also in objective before the :return, and in vlmop2 and dtlz1). This currently produces an invisible warning in the docs build, but since these warnings sometimes result in bad documentation we're planning on turning them into errors that get checked as part of the pre-merge quality checks (and which can be run locally using tox).

Reply:
The blank line has been added.